src/scene_battle.cpp

```diff
@@ -163,17 +158,28 @@ void Scene_Battle::Update() {
 		Scene::Pop();
 	}
 
-	bool events_finished = Game_Battle::UpdateEvents();
+	do {
+		bool events_finished = Game_Battle::UpdateEvents();
+		changed_state = state;
 
-	if (Game_Temp::gameover) {
-		Game_Temp::gameover = false;
-		Scene::Push(std::make_shared<Scene_Gameover>());
-	}
+		if (Game_Temp::gameover) {
+			Game_Temp::gameover = false;
+			Scene::Push(std::make_shared<Scene_Gameover>());
+		}
 
-	if (!Game_Message::visible && events_finished) {
-		ProcessActions();
-		ProcessInput();
-	}
+		if (!Game_Message::visible && events_finished) {
+			options_window->Update();
+			status_window->Update();
+			command_window->Update();
+			help_window->Update();
+			item_window->Update();
+			skill_window->Update();
+			target_window->Update();
+
+			ProcessActions();
+			ProcessInput();
+		}
+	} while (changed_state != state && state == State_SelectOption);
```

Comment:
> 2.1: RM2000: If no hero party is ready to attack and they can't choo

Uhm I need to test around to figure out what this is supposed to do

Reply:
To reproduce the bug, create a troop page on turn 0 that sets a state like sleep to your entire party. When the battle starts, the battle menu will appear for a few frames and then disappear.

Reply:
.. and testing it with this branch.. the bug appears to not be fixed. I think this one needs a refactor. Unless I misinterpreted the commit message?